### PR TITLE
Swatches options: eliminate objects instantiation since only their data needed

### PR DESCRIPTION
**Pull request by ilnytskyi**
### Description (*)

Removed the need to instantiate new objects from collection in the method
`\Magento\Swatches\Helper\Data::getSwatchesByOptionsId`
since only data used

Objects are created but then on every used a method `->getData`
Selection_380

With this fix we only use data and do not create new objects, that saves a few milliseconds on large collections
Selection_379

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. Render layered navigation on listing
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@ilnytskyi In general your change looks logical, but it prevents triggering all events, so it might break some customizations. Let's wait for test results

**Comment by ilnytskyi** [comment]
@ihor-sviziev 
I agree if there is native magneto customization.
To customize that method or collection there are plenty of ways, from plugins to injecting custom factory collection
I will take a look at falling tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please fix failing tests

File: app/code/Magento/Swatches/Test/Unit/Helper/DataTest.php [hunk 1]
```diff
@@ -737,28 +735,21 @@ public function testGetSwatchesByOptionsIdIf2()
...
-
-        $swatchCollectionMock = $this->objectManager->getCollectionMock(
-            Collection::class,
-            [
-                $swatchMock,
-                $swatchMock,
-            ]
-        );
+        // @codingStandardsIgnoreStart
+        $swatchMock->expects($this->at(0))->method('offsetGet')->with('type')->willReturn($optionsData[0]['type']);
```
Comment (ihor-sviziev) [inline app/code/Magento/Swatches/Test/Unit/Helper/DataTest.php:739]: @ilnytskyi maybe it's better to replace it with `willReturnMap`?

Comment (ihor-sviziev) [reply app/code/Magento/Swatches/Test/Unit/Helper/DataTest.php:739]: ah, no, it will not work, as values the same. So please leave it as is

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6922](https://jira.corp.magento.com/browse/ENGCOM-6922) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**